test/run

```diff
@@ -195,60 +217,104 @@ test_from_dockerfile(){
   check_result $?
 }
 
-# Since we built the candidate image locally, we don't want S2I attempt to pull
-# it from Docker hub
-s2i_args="--pull-policy=never"
+test_application_with_user() {
+  # test application with random user
+  CONTAINER_ARGS="--user 12345" test_application
 
-# Verify the 'usage' script is working properly when running the base image with 's2i usage ...'
-test_s2i_usage
-check_result $?
+}
 
-# Verify the 'usage' script is working properly when running the base image with 'docker run ...'
-test_docker_run_usage
-check_result $?
+test_application_enable_init_wrapper() {
+  # test application with init wrapper
+  CONTAINER_ARGS="-e ENABLE_INIT_WRAPPER=true" test_application
+}
+
+test_scl_variables_in_dockerfile() {
+  if [ "$OS" == "rhel7" ] || [ "$OS" == "centos7" ]; then
+     # autocleanup only enabled here as only the following tests so far use it
+     CID_FILE_DIR=$(mktemp -d)
+     ct_enable_cleanup
 
+     info "Testing variable presence during \`docker exec\`"
+     ct_check_exec_env_vars
+     check_result $?
```

Comment:
If the `check_result` here returns 1, does it terminate the function or not? And if not, will this result be reflected in the final summary?

Reply:
Yes, if `check_result` returns 1, it sets a variable `TESTCASE_RESULT`, and based on the value it marks the tests as a failure.